test/auth/auth.test.js

```diff
@@ -0,0 +1,62 @@
+/* eslint-env mocha */
+import proclaim from 'proclaim';
+import fetchMock from 'fetch-mock';
+
+import {getJsonWebToken} from '../..//src/js/utils/auth';
+
+describe("Auth", () => {
+	describe("getJsonWebToken", () => {
+		it("is a function", () => {
+			proclaim.isFunction(getJsonWebToken);
+		});
+
+		describe("when comments auth service returns a valid response", () => {
+			before(() => {
+				fetchMock.mock('https://comments-auth.ft.com/v1/jwt/', {
+					token: '12345'
+				});
+			});
+
+			after(() => {
+				fetchMock.reset();
+			});
+
+			it("returns a promise", () => {
+				const returnValue = getJsonWebToken();
+				proclaim.isInstanceOf(returnValue, Promise);
+			});
+
+			it("returns a promise which contains a JSON Web Token as a string", () => {
+				getJsonWebToken()
+					.then(token => proclaim.isString(token));
+			});
+		});
+		describe("when the comments auth service response is missing the token", () => {
+			before(() => {
+				fetchMock.mock('https://comments-auth.ft.com/v1/jwt/', {});
+			});
+
+			after(() => {
+				fetchMock.reset();
+			});
+
+			it("throws an error", () => {
+				proclaim.throws(getJsonWebToken());
```

Comment:
Not sure how proclaim works, but shouldn't this (and the other bad scenario) be tested for a rejected promise? This test is apparently passing, but I just couldn't make sense of it.

Reply:
You are right this is not testing how I thought it was I will investigate

Reply:
`proclaim.throws` takes a function and will call that function and ensure that it does throw an error.
```suggestion
				proclaim.throws(function() {
					getJsonWebToken();
				});
```

Reply:
This is what I tried originally and it didn't work which I think is how I ended up with this current code. As it was already a function I thought "Ahh I don't need to wrap it" and it passed and in my excitement I forgot to test that it passing for the right reasons.

Wrapping it in a function seems to result in it failing, I am guessing it might have something to do with it being inside a promise.

Reply:
Yes, that makes sense, proclaim is not async/promise aware. We should either move to a more maintained assertion library which supports promises or we should look to add promise support to proclaim.

Reply:
> We should either move to a more maintained assertion library

Did you hear that @rowanmanning ? I think Jake just called you lazy.

![tenor](https://user-images.githubusercontent.com/524573/57029052-95e08b00-6c38-11e9-9b65-117e5739c6d5.gif)

On a serious note, yeah I am looking at the alternatives at the moment

Reply:
You can use proclaim with promises and `throws` by using async function with await:
```
proclaim.throws(async function() {
 await getJsonWebToken();
});
```
The await will turn a promise rejection into a thrown error